### DISCUSSION TITLE
Installing Breeze Qt widget style

### DISCUSSION THREAD

**Discussion by edwardp3** [lxqt/lxqt#2247] [Q&A] [answered]
Are certain packages required to install the Breeze Qt widget style? I am now using Mageia (8) and like other distros offering LXQt, it has Fusion and Windows for the widget styles. 

I am aware through one of my past threads that there is an issue with Fusion where it would not display the LXQt bird icon on the main menu with the System theme, but changing the widget style to Windows then displays the icon.

Thanks in advance.

**Comment by tsujan** [comment, marked as answer]
> Are certain packages required to install the Breeze Qt widget style?

Yes. In Debian its name is "kde-style-breeze", in Arch it's just "breeze", and in other distros it may have other names, but they all contain "breeze". You need to find its name in your distro and install it. Then you could see and activate it in LXQt Appearance Configuration → Widget Style → Qt Style.

> I am aware through one of my past threads that there is an issue with Fusion where it would not display the LXQt bird icon on the main menu with the System theme,

It wasn't Fusion's fault but a small problem in the System theme, which was fixed.

**Reply by edwardp3** [reply]
Hi Tsujan, thanks for the reply. There are several packages with 'breeze' in the name, with one just named 'breeze', I will look at that.

Re the System theme, I have extracted the themes from the themes package here into /.local/share/lxqt/themes and when I select the System theme (with Fusion widgets) and uncheck the main menu icon (which is the Mageia logo by default), the LXQt bird logo isn't displayed, the space is empty, it only displays if Windows widgets is selected. So when I change themes, it may be loading them from the system directory instead (/usr/share/lxqt/themes) if there are duplicates, rather than from /.local/share/lxqt/themes. There is a theme included with Mageia that is not in the .ZIP file: "Artemis" and themes in the .ZIP not included with Mageia, such as Arch-Colors, silver and Valendas, but when on the Appearance Configuration screen, I see all themes from both directories, but any duplicates such as System are listed only once. As the System theme is shown as "system" in /usr/local..., I don't know if changing the theme in /.local/share/lxqt... capitalizing "System" will make a difference.

Mageia 8 was released in February 2021, so the installation has the packages that were released at that time. The LXQt version is 0.16.0.

EDIT: Changing "system" to "System" in /.local/share caused "System" to be listed twice, with only the original LXQt bird graphic and the second with the bird graphic and the desktop numbers. Selecting the entry displaying the desktop numbers caused the main menu icon to then display the LXQt bird logo.

**Reply by tsujan** [reply]
> EDIT: Changing "system" to "System" in /.local/share caused "System" to be listed twice...

Yes, one is the root theme, the other is the user theme.

We have a plan to add `index.theme` to each theme, so that their names could be changed without renaming their folders.

> The LXQt version is 0.16.0.

I'm afraid it's very old. Lots of bugs have been fixed and many features have been added since then. The latest release is 1.1.0.

**Reply by edwardp3** [reply]
Basically, capitalizing "System" in the /.local directory, achieved the desired result.  :)

The next version of Mageia, currently in testing, is showing 1.1.0 in the 'testing' repository with 1.0.0 in the release repository, so I think once it becomes Mageia 9 and is released, I trust the upgrade will install 1.1.0.

**Reply by edwardp3** [reply]
In Mageia 8, installing the "breeze" package also installed relevant other dependencies. It's all set. Thank you.

**Reply by edwardp3** [reply]
_I guess the subject cannot be changed once it's submitted, so..._

For the benefit of others, in Mageia, the 'oxygen' package installed the Oxygen widget style used by LXQt.

**Reply by tsujan** [reply]
> the 'oxygen' package installed the Oxygen widget style used by LXQt.

Qt only provides Fusion and the ugly Windows styles. All other styles should be installed separately. That's independent of LXQt and other DEs.

**Reply by edwardp3** [reply]
> Qt only provides Fusion and the ugly Windows styles. All other styles should be installed separately. That's independent of LXQt and other DEs.

I looked at the Windows style, I'm in agreement. I didn't like it. I'm currently using the Oxygen widget and icon themes.

**Reply by tsujan** [reply]
A list of well-known Qt styles:

 * Fusion, Qt's default style
 * Breeze and Oxygen, belonging to KDE.
 * QtCurve. A nice and very flexible Qt style with lots of options.
 * Kvantum. An SVG-based Qt style, which is the most flexible and has tons of options.
 * Cleanlooks, Plastique and Gtk2 (not related to GTK+ itself), belonging to `qt5-styleplugins`. Sadly, it can't be recommended because It should be recompiled every now and then; otherwise, its styles will cause immediate crashes in all Qt apps :(

There are other good styles too.

Currently, among the above-mentioned styles, only Fusion and Kvantum work with Qt6, but Breeze and QtCurve might be updated soon (I don't know about Oxygen).

**Reply by edwardp3** [reply]
Thank you for the other information. Kvantum is not available with Mageia, but QtCurve is, so I installed that, am trying it out now.

**Comment by AzumaHazuki** [comment]
Sorry for the long bump here, but I promise this is actually relevant: the old Plastik style (you know, from way back in the KDE 3.5.x days?) has actually been ported to Qt5.x and 6.x! Details at <a href=https://www.pling.com/p/1931943>pling.com</a>.

I've also made a Plastik LXQt theme and an Openbox theme that gets...well, as close to the old Plastik windeco as can reasonably be expected; screenshot <a href=https://i.imgur.com/6Pnlr5Y.jpeg>here</a>.

From preliminary experiments, the Qt 5/6 port of Plastik plays well with LXQt's 8-color palettes :) I do wonder, could we possibly include support for KDE's color schemes? Or is KDE breaking Qt somehow and forcing those schemes' colors onto bits of Qt that normally can't be themed?

**Reply by tsujan** [reply]
> could we possibly include support for KDE's color schemes?

KDE's color schemes are more than Qt palettes and depend on KDE codes. In a sense, they are settings for KDE's widgets styles, Breeze and Oxygen. So, no, LXQt can't support them.

> Or is KDE breaking Qt somehow and forcing those schemes' colors onto bits of Qt that normally can't be themed?

It can be said that every style "forces" some things on Qt; that's normal. But LXQt shouldn't depend on extra codes needed for that.

> I've also made a Plastik LXQt theme

You're very welcome to make a PR at https://github.com/lxqt/lxqt-themes.

**Comment by AzumaHazuki** [comment]
I'd like to make a few more themes first I think...would you be okay with an Adwaita look, for example?

Also, regarding the color schemes, since LXQt uses only 8 colors I might be able to write a simple bash script to match them to the relevant 8 entries in a given KDE scheme and extract them into LXQt-compatible palette files :)

**Reply by tsujan** [reply]
> would you be okay with an Adwaita look, for example?

Why not? Different themes for different tastes are welcome.

**Reply by TheESN** [reply]
> Also, regarding the color schemes, since LXQt uses only 8 colors I might be able to write a simple bash script to match them to the relevant 8 entries in a given KDE scheme and extract them into LXQt-compatible palette files :)

That means that you can modify these themes to be fully LXQt's palettes compatible?

**Reply by AzumaHazuki** [reply]
Unfortunately no. LXQt palettes, at present, have only 8 colors and their variable names are different from what's in a KDE5 scheme, for example.

Here's what the inside of the Clearlooks palette I made looks like:
```

[Palette]
base_color=#ffffff
highlight_color=#80a0d0
highlighted_text_color=#ffffff
link_color=#0000ee
link_visited_color=#ffb3f7
text_color=#1a1a1a
window_color=#edeceb
window_text_color=#1a1a1a
```

And here's what a Qogir color scheme for Plasma looks like:
```
[ColorEffects:Disabled]
Color=112,111,110
ColorAmount=0
ColorEffect=0
ContrastAmount=0.65
ContrastEffect=1
IntensityAmount=0.1
IntensityEffect=0

[ColorEffects:Inactive]
ChangeSelectionColor=true
Color=112,111,110
ColorAmount=0.025
ColorEffect=2
ContrastAmount=0.1
ContrastEffect=2
Enable=true
IntensityAmount=0
IntensityEffect=0

[Colors:Button]
BackgroundAlternate=251,251,252
BackgroundNormal=255,255,255
DecorationFocus=82,148,226
DecorationHover=82,148,226
ForegroundActive=255,128,224
ForegroundInactive=136,135,134
ForegroundLink=0,87,174
ForegroundNegative=191,3,3
ForegroundNeutral=176,128,0
ForegroundNormal=49,54,59
ForegroundPositive=0,110,40
ForegroundVisited=69,40,134

[Colors:Complementary]
BackgroundAlternate=50,52,61
BackgroundNormal=40,42,51
DecorationFocus=82,148,226
DecorationHover=82,148,226
ForegroundActive=61,174,233
ForegroundInactive=102,106,115
ForegroundLink=41,128,185
ForegroundNegative=218,68,83
ForegroundNeutral=246,116,0
ForegroundNormal=230,235,239
ForegroundPositive=39,174,96
ForegroundVisited=82,148,226

[Colors:Header]
BackgroundAlternate=50,52,61
BackgroundNormal=40,42,51
DecorationFocus=82,148,226
DecorationHover=82,148,226
ForegroundActive=61,174,233
ForegroundInactive=102,106,115
ForegroundLink=41,128,185
ForegroundNegative=218,68,83
ForegroundNeutral=246,116,0
ForegroundNormal=230,235,239
ForegroundPositive=39,174,96
ForegroundVisited=82,148,226

[Colors:Header][Inactive]
BackgroundAlternate=50,52,61
BackgroundNormal=40,42,51
DecorationFocus=82,148,226
DecorationHover=82,148,226
ForegroundActive=61,174,233,150
ForegroundInactive=102,106,115
ForegroundLink=41,128,185
ForegroundNegative=218,68,83
ForegroundNeutral=246,116,0
ForegroundNormal=230,235,239,150
ForegroundPositive=39,174,96
ForegroundVisited=82,148,226

[Colors:Selection]
BackgroundAlternate=82,148,226
BackgroundNormal=82,148,226
DecorationFocus=93,169,255
DecorationHover=93,169,255
ForegroundActive=255,128,224
ForegroundInactive=142,203,233
ForegroundLink=0,87,174
ForegroundNegative=156,14,14
ForegroundNeutral=255,221,0
ForegroundNormal=250,251,252
ForegroundPositive=128,255,128
ForegroundVisited=69,40,134

[Colors:Tooltip]
BackgroundAlternate=50,52,61
BackgroundNormal=40,42,51
DecorationFocus=82,148,226
DecorationHover=82,148,226
ForegroundActive=255,128,224
ForegroundInactive=96,112,128
ForegroundLink=0,87,174
ForegroundNegative=191,3,3
ForegroundNeutral=176,128,0
ForegroundNormal=211,218,227
ForegroundPositive=0,110,40
ForegroundVisited=69,40,134

[Colors:View]
BackgroundAlternate=248,248,248
BackgroundNormal=250,251,252
DecorationFocus=82,148,226
DecorationHover=82,148,226
ForegroundActive=255,128,224
ForegroundInactive=136,135,134
ForegroundLink=0,87,174
ForegroundNegative=191,3,3
ForegroundNeutral=176,128,0
ForegroundNormal=49,54,59
ForegroundPositive=0,110,40
ForegroundVisited=69,40,134

[Colors:Window]
BackgroundAlternate=242,242,242
BackgroundNormal=255,255,255
DecorationFocus=82,148,226
DecorationHover=82,148,226
ForegroundActive=255,128,224
ForegroundInactive=136,135,134
ForegroundLink=0,87,174
ForegroundNegative=191,3,3
ForegroundNeutral=176,128,0
ForegroundNormal=49,54,59
ForegroundPositive=0,110,40
ForegroundVisited=69,40,134

[General]
ColorScheme=Qogir
Name=Qogir
shadeSortColumn=true

[KDE]
contrast=0

[WM]
activeBackground=40,42,51
activeBlend=50,52,61
activeForeground=211,218,227
inactiveBackground=40,42,51
inactiveBlend=50,52,61
inactiveForeground=120,124,129
```

That said, if we can figure out which 8 variables in these schemes correspond to which ones in an LXQt palette, extracting them should be fairly straightforward with grep, cut, and so on...

**Reply by AzumaHazuki** [reply]
Ooh, I just found something interesting: the internal format of a Kvantum theme has very, very similar color variable names to an LXQt scheme :) Since IIRC Tsujan created Kvantum that's not too surprising, but this is excellent news. 

Compare and contrast the Colors section of Materia.Dark.kvconfig with an LXQt palette:
```
[GeneralColors]
alt.base.color=#1e1e1e
base.color=#1e1e1e
button.color=#2e2e2e
button.text.color=#dfdfdf
dark.color=#0d0d0d
disabled.text.color=#818181
highlight.color=#8ab4f83c
highlight.text.color=white
inactive.highlight.color=#8ab4f83c
light.color=#484848
link.color=#8ab4f8
link.visited.color=#ce93d8
mid.color=#2e2e2e
mid.light.color=#ffffff1e
progress.indicator.text.color=#dfdfdf
text.color=#dfdfdf
tooltip.text.color=white
window.color=#121212
window.text.color=#dfdfdf
```

If Kvantum becomes more tightly integrated with LXQt, a file-watching hook that pays attention to newly-installed themes and autogenerates a palette for them might be a possibility. 

I know we're trying to keep LXQt as light and dependency-free as possible, but it may be a good idea to depend on and default to Kvantum for the Qt configuration someday for the sake of a consistent appearance... @tsujan this would be waaay in the future probably but what do you think?

**Reply by tsujan** [reply]
> it may be a good idea to depend on and default to Kvantum for the Qt configuration someday

I started Kvantum before knowing about LXQt. Later, I made it tuned to LXQt (and KDE), but it's fully DE-independent. I've never made a change to LXQt because of Kvantum. LXQt's palette is just a part of Qt's *standard* palette.

IMO, we should never make LXQt depend on anything artificially. All Qt widget styles can be used with LXQt. I use Kvantum, another user prefers Breeze, yet another user likes Fusion, etc.

The colors set under `GeneralColors` are the most basic colors Kvantum uses; the other colors are in the SVG files of Kvantum themes and can have complex gradients. In other words, unlike Breeze and Fusion, Kvantum's general colors depend on the theme and can't be edited freely.

**Reply by tsujan** [reply]
Oh, I forgot to tell this:

Kvantum is among those styles that don't follow the palette defined by the DE, whether LXQt or KDE (Qt itself allows that). If you change LXQ's palette, Fusion and Breeze will be affected; Kvantum won't.

So, in this sense, Kvantum is the "worst" choice for the *default* style of LXQt.

IMHO, we should stick with Fusion, Qt's default style. Users are free to change it for themselves.

**Reply by TheESN** [reply]
The problem is that some themes like Breeze can look artificially compared to other themes in some cases, such as choosing dark LXQt palette, which cause consistency problems and using Fusion is the only solution. I got this problem first in Lubuntu where such themes are included by standard, what really confused me at the moment

**Reply by AzumaHazuki** [reply]
Good point, @tsujan ...not really much we can do about this, theming LXQt is made of a lot of moving parts, huh? @_@

I've been leveraging the ability of .qss stylesheets to pull from LXQt palettes and have created <a href=https://github.com/AzumaHazuki/lxqt-themes-arc>Arc</a>, <a href=https://github.com/AzumaHazuki/lxqt-themes-materia>Materia</a>, and <a href=https://github.com/AzumaHazuki/lxqt-themes-mint-y>Mint-Y</a> theme sets. 

Probably none of these reeeeally belong in the base theme set, I should do an Adwaita/Adwaita-Dark pair for that, but it's interesting how the bits can be brought in line with one another...
